samples_project/Assets/SampleViewer/Samples/FeatureLayer/FeatureLayer.cs

```diff
@@ -0,0 +1,421 @@
+            var coordinates = feature.SelectToken("geometry").SelectToken("coordinates").ToArray();
+            var properties = feature.SelectToken("properties").ToArray();
+
+            if (GetAllOutfields)
+            {
+                foreach (var value in properties)
+                {
+                    var key = value.ToString();
+                    var props = key.Split(":");
+                    currentFeature.properties.propertyNames.Add(props[0]);
+                    currentFeature.properties.data.Add(props[1]);
+                    featureInfo.Properties.Add(props[1]);
+                }
+            }
+            else
+            {
+                for (var j = 0; j < outfields.Count; j++)
+                {
+                    if (OutfieldsToGet.Contains(outfields[j]))
+                    {
+                        var key = properties[j].ToString();
+                        var props = key.Split(":");
+                        currentFeature.properties.propertyNames.Add(props[0]);
+                        currentFeature.properties.data.Add(props[1]);
+                        featureInfo.Properties.Add(props[1]);
+                    }
+                }
+            }
+
+            foreach (var coordinate in coordinates)
+            {
+                currentFeature.geometry.coordinates.Add(Convert.ToDouble(coordinate));
+                featureInfo.Coordinates.Add(Convert.ToDouble(coordinate));
+            }
+
+            featureInfo.ArcGISCamera = arcGISCamera;
+            ArcGISPoint Position = new ArcGISPoint(featureInfo.Coordinates[0], featureInfo.Coordinates[1],
+                stadiumSpawnHeight, new ArcGISSpatialReference(featureSRWKID));
+            ArcGISRotation Rotation = new ArcGISRotation(0.0, 90.0, 0.0);
+            locationComponent.enabled = true;
+            locationComponent.Position = Position;
+            locationComponent.Rotation = Rotation;
+            Features.Add(currentFeature);
+            FeatureItems.Add(featureItem);
+        }
+    }
+
+    private void CreateFeatureRange()
+    {
+        if (jFeatures.Length < LastValue)
+        {
+            UIManager.DisplayText = FeatureLayerUIManager.TextToDisplay.IndexOutOfBoundsError;
+            for (int i = StartValue; i < jFeatures.Length; i++)
+            {
+                Feature currentFeature = new Feature();
+                var featureItem = Instantiate(featurePrefab, this.transform);
+                featureItem.tag = "FeatureItem";
+                featureItem.layer = 7;
+                featureInfo = featureItem.GetComponent<FeatureData>();
+                locationComponent = featureItem.GetComponent<ArcGISLocationComponent>();
+                var coordinates = jFeatures[i].SelectToken("geometry").SelectToken("coordinates").ToArray();
+                var properties = jFeatures[i].SelectToken("properties").ToArray();
+
+                if (GetAllOutfields)
```

Comment:
I'm concerned by the amount of repeat code here. Is there a way to refactor?

Reply:
I'm not sure of a way to avoid the repeating code. It is the same in unreal. If we are going to have the ability to get all and get only selected, we basically need to repeat the code but in different loops. One being a for each loop and one being a for loop with the first and last values being the user specified ones. The third block that is the same is a safety net to avoid console errors if they put a higher max value than there are features in the layer.

Reply:
Why not have get all call the index method and provide the first and last index as arguments?